### PR TITLE
MPS: Fixes

### PR DESCRIPTION
**Pull request by kulinseth**
Cast integer to float in UnaryOps
Add tensor dtype in key generation
Enable FP16 scalars and use placeholder for alpha tensor in add/sum ops

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: aten/src/ATen/native/mps/OperationUtils.mm [hunk 1]
```diff
@@ -71,15 +72,18 @@
...
+              ":" + getMPSTypeString(self.scalar_type());
 }
 
 void runMPSGraph(
     MPSStream* mpsStream,
     MPSGraph* mpsGraph,
     NSDictionary* feeds,
     NSDictionary* results) {
-
+#ifdef DEBUG
```
Comment (albanD) [inline aten/src/ATen/native/mps/OperationUtils.mm:84]: Do we really want to unconditionally print here?

**Review by albanD — APPROVED** [review]
SGTM!
We should remove the extra print though I think before merging

**Comment by albanD** [comment]
@razarmehr is this last commit expected to be in this PR? Why is this removed? And if it shouldn't be there, we should remove the surrounding code.

**Comment by kulinseth** [comment]
> @razarmehr is this last commit expected to be in this PR? Why is this removed? And if it shouldn't be there, we should remove the surrounding code.

@albanD , the last commit is due to a big perf regression we noticed as a result of recreating graphs. The supporting code for callbacks is good, so I have removed surrounding code as requested.

**Review by albanD — APPROVED** [review]
ok

**Comment by malfet** [comment]
Please do not pick this one into 1.12.0, as it seems to re-introduce initialization time dependency to MPS.

**Pull request closed without merging**